### PR TITLE
fix: flaky test `Increase Token Allowance increases token spending ca..`

### PR DESCRIPTION
**Pull request by hjetpoluru**
## **Description**

This PR fixes the flaky test 

[Open in GitHub Codespaces](https://codespaces.new/MetaMask/metamask-extension/pull/26640?quickstart=1)

## **Related issues**

Fixes:
https://github.com/MetaMask/metamask-extension/issues/26438

## **Manual testing steps**

To execute the test locally or in codespaces
yarn
yarn build:test:webpack
ENABLE_MV3=false yarn test:e2e:single test/e2e/tests/tokens/increase-token-allowance.spec.js --browser=chrome

Validate the test should pass in CI 

## **Pre-merge author checklist**

- [ ] I've followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Extension Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: test/e2e/tests/tokens/increase-token-allowance.spec.js [hunk 1]
```diff
@@ -283,7 +280,7 @@ describe('Increase Token Allowance', function () {
       text: `${finalSpendingCap} TST`,
       css: '.mm-box > h6',
     });
-    await driver.clickElement({
+    await driver.clickElementAndWaitForWindowToClose({
```
Comment (hjetpoluru) [inline test/e2e/tests/tokens/increase-token-allowance.spec.js:283]: This is the main change where the dialog needs to be closed too.

Comment (HowardBraham) [reply test/e2e/tests/tokens/increase-token-allowance.spec.js:283]: Yay, you're using David's new function!

File: test/e2e/tests/tokens/increase-token-allowance.spec.js [hunk 2]
```diff
@@ -296,6 +293,8 @@ describe('Increase Token Allowance', function () {
       css: '.transaction-list__completed-transactions .activity-list-item [data-testid="activity-list-item-action"]',
       text: 'Increase TST spending cap',
     });
+
+    await driver.delay(2000);
```
Comment (HowardBraham) [inline test/e2e/tests/tokens/increase-token-allowance.spec.js:297]: Is there an alternative possible?  What are we waiting for here?

File: test/e2e/tests/tokens/increase-token-allowance.spec.js (hunk 1 above)
Comment (hjetpoluru) [reply test/e2e/tests/tokens/increase-token-allowance.spec.js:283]: yeah that's right!!

File: test/e2e/tests/tokens/increase-token-allowance.spec.js (hunk 2 above)
Comment (hjetpoluru) [reply test/e2e/tests/tokens/increase-token-allowance.spec.js:297]: Another alternative would be to check if the status in the extension is updated, which is already mentioned above the delay. However we still require a small delay, so it should be fine.

**Review by HowardBraham — APPROVED** [review]

**Review by seaona — APPROVED** [review]
LGTM!

**Pull request merged**